drivers/usb/device/usb_dc_stm32.c

```diff
@@ -511,18 +511,39 @@ int usb_dc_ep_read(const u8_t ep, u8_t *const data, const u32_t max_data_len,
 		ep_state->read_offset += read_count;
 	}
 
+done:
+	if (read_bytes) {
+		*read_bytes = read_count;
+	}
+
+	return 0;
+}
+
+int usb_dc_ep_read_continue(u8_t ep)
+{
+	struct usb_dc_stm32_ep_state *ep_state = usb_dc_stm32_get_ep_state(ep);
+
+	if ((ep == EP0_OUT) || !EP_IS_OUT(ep)) { /* Check if OUT ep */
+		SYS_LOG_ERR("Not valid endpoint: %02x", ep);
+		return -EINVAL;
+	}
+
 	/* If no more data in the buffer, start a new read transaction.
 	 * DataOutStageCallback will called on transaction complete.
 	 */
-	if (ep != EP0_OUT && !ep_state->read_count) {
+	if (!ep_state->read_count) {
 		usb_dc_ep_start_read(ep, usb_dc_stm32_state.ep_buf[EP_IDX(ep)],
 				     USB_OTG_FS_MAX_PACKET_SIZE);
 	}
 
-done:
-	if (read_bytes) {
-		*read_bytes = read_count;
-	}
+	return 0;
+}
+
+int usb_dc_ep_read(const u8_t ep, u8_t *const data, const u32_t max_data_len,
+		   u32_t * const read_bytes)
+{
+	usb_dc_ep_read_wait(ep, data, max_data_len, read_bytes);
+	usb_dc_ep_read_continue(ep);
 
```

Comment:
Could you please check returned values. Now it's always 0 (for now).

Reply:
the previous version always return 0, so there i haven't change it.
you means there need return negative when fails, like usb_dc_dw.c did?

Reply:
I mean usb_dc_ep_read_wait and usb_dc_ep_read_continue method return value we should not ignore.

Reply:
I had fix this, please review again.